src/custom/pages/Claim/index.tsx

```diff
@@ -242,8 +250,14 @@ export default function Claim() {
       <InvestmentFlow
         isAirdropOnly={isAirdropOnly}
         hasClaims={hasClaims}
-        approveCallback={approveCallback}
-        approveState={approveState}
+        gnoApproveData={{
+          approveCallback: gnoApproveCallback,
+          approveState: gnoApproveState,
+        }}
+        usdcApproveData={{
+          approveCallback: usdcApproveCallback,
+          approveState: usdcApproveState,
+        }}
```

Comment:
This is fine for now, but I'd rather have the approval logic inside the claim component.
We don't need to have this at the index, or do we?

Also, I still think we can do a generic approval flow rather than one for usdc and one for gno always even when they are not used.

Refactor for another day I guess

Reply:
probably easier, just didnt like "mapping" components having this logic